Feat(payment): 결제 도메인 v3 기반 아웃박스패턴 + rabbitmq 구현

ZeroColaa opened the pull request:
###  주요 변경사항 

#### 1. 결제 이벤트 아키텍처 개선
- Outbox 테이블(payment_outbox) 기반 이벤트 저장
- @TransactionalEventListener 기반 MQ 발행 구조 적용
- 결제 성공/실패/환불 시 Completed/Failed/Refunded 이벤트 발행

#### 2. 이벤트 재처리 및 논리적 DLQ 처리 개선
- 미발행 이벤트 재전송 스케줄러 (PENDING 상태 대상)
- 실패 이벤트 재시도 스케줄러 (FAILED && retryCount < 3)
- 3회 이상 실패한 이벤트 논리적 DLQ 처리 (status = DEAD_LETTERED)

### 참고 
- 취소(CANCELED)는 단순 사용자 중단 처리이므로 이벤트 미발행
- 모임과 연결 되는 부분은 오늘 중 구현할 예정입니다
- 네이밍이나 패키지 컨벤션 올려주신 부분도 오늘 중 수정할 예정입니다

ZeroColaa committed:
feat(rabbitmq): Payment 로직 수정관련 전용 Rabbitmq 설정 클래스

ZeroColaa committed:
Merge branch 'dev-v3' of https://github.com/momo-spring-project/momo-backend into feat/payment-v3

ZeroColaa committed:
feat(payment): Outbox 엔티티 및 DIP 기반 리포지토리 구현 및 마이그레이션 추가

ZeroColaa committed:
feat(payment): Outbox 발행/재시도/정리 스케줄러 및 서비스 구현

ZeroColaa committed:
feat(payment): RabbitMQ DTO, 이벤트 Producer/Consumer 구현

ZeroColaa committed:
refactor(payment): 패키지 구조 변경 및 불필요 클래스 제거

ZeroColaa committed:
refactor(payment): Outbox 및 rabbitmq 기반 결제 로직으로 전환

ZeroColaa committed:
feat(payment): 참가자 생성 이벤트 수신 시 결제 처리 및 이벤트 발행 추가

- meeting.participant.created 이벤트 수신 시 결제 진행(선행 구현)
- 결제 성공/실패 결과에 따라 PaymentCompleted/PaymentFailed 이벤트를 RabbitMQ로 발행

hojin915 commented on src/main/java/com/example/momo/domain/payment/application/PaymentServiceImpl.java:
```diff
@@ -40,64 +44,108 @@
+		}
+
 		// 4. 무료 모임인 경우 별도 처리
 		if (amount == 0) {
```
지금 생각해보니까 무료인 경우 결제를 요청하는 도메인에서 결제 도메인까지 안넘어오도록 하는게 맞는 것 같은데요
모임 참가쪽에서 필터링 하고 결제 도메인에서는 안전장치의 의미로 결제 흐름만 끊는건 어떤지 제안드립니다

ZeroColaa replied:
알겠습니다 감사합니다 우선은 머지 후 모임쪽과 연결할 때 고쳐놓겠습니다!

hojin915 reviewed (approved):
고생하셨습니다

Mybread2 reviewed (approved):
수고하셨습니다...

ZeroColaa committed:
feat(payment): Payment 엔티티 생성자 접근 제한 및 소비자 로직 수정

- Payment 엔티티에 @AllArgsConstructor 제거
- PaymentEventConsumer 주석 변경및 트랜잭션 어노테이션 제거
- 마이그레이션 파일 V13__Create_payment_outbox_table.sql 로 이름 변경

ZeroColaa committed:
Merge branch 'dev-v3' of https://github.com/momo-spring-project/momo-backend into feat/payment-v3
